src/Front/Pattern/Desugar.hs

```diff
@@ -341,6 +341,7 @@ substId old new = substExpr
     in  A.Match (substExpr e) (map substArm arms)
   substExpr e@(A.CQuote _  ) = e
   substExpr (  A.CCall s es) = A.CCall s $ map substExpr es
+  substExpr (A.ListExpr es) = A.ListExpr $ map substExpr es
```

Comment:
This is causing redundant pattern matches. I think you can delete this line because you already have this on line 309?